flink-formats/flink-parquet/src/test/java/org/apache/flink/formats/parquet/row/ParquetRowDataWriterTest.java

```diff
@@ -120,15 +144,18 @@ private void innerTest(Configuration conf, boolean utcTimestamp) throws IOExcept
                             toDateTime(v),
                             BigDecimal.valueOf(v),
                             BigDecimal.valueOf(v),
-                            BigDecimal.valueOf(v)));
+                            BigDecimal.valueOf(v),
+                            new Integer[] {v},
+                            mapData,
+                            Row.of(String.valueOf(v), v)));
         }
 
         ParquetWriterFactory<RowData> factory =
-                ParquetRowDataBuilder.createWriterFactory(ROW_TYPE, conf, utcTimestamp);
+                ParquetRowDataBuilder.createWriterFactory(ROW_TYPE_COMPLEX, conf, utcTimestamp);
         BulkWriter<RowData> writer =
                 factory.create(path.getFileSystem().create(path, FileSystem.WriteMode.OVERWRITE));
         for (int i = 0; i < number; i++) {
-            writer.addElement(CONVERTER.toInternal(rows.get(i)));
+            writer.addElement(CONVERTER_COMPLEX.toInternal(rows.get(i)));
         }
         writer.flush();
         writer.finish();
```

Comment:
I think you need add assert below. No validation for written data.
Maybe you can use something like `ParquetAvroStreamingFileSinkITCase.readParquetFile`.

Reply:
add test function `complexTypeTest`